### DISCUSSION TITLE
web3modal + wagmi - Send a custom json rpc request to the wallet (EIP-1193)

### DISCUSSION THREAD

**Discussion by Sednaoui** [WalletConnect/walletconnect-monorepo#3049] [Developer Support]
Hello,

I am looking to get the provider object on web3modal (specifically to a wallet connected through walletconnect and not `window.ethereum`), so I can pass it to wagmi's [createWalletClient](https://viem.sh/docs/clients/wallet.html#wallet-client) like the following:

```js

const web3modalProvider = ??

const walletClient = createWalletClient({
 chain: mainnet,
 transport: custom(web3modalprovider)
})
```

My usecase is sending a custom wallet request (EIP-1193) to the wallet connected through walletconnect

```js
const sendRequest = await walletClient.request({
  method: "custom_request",
  params: [args],
});
```

**Comment by tcheee** [comment]
Hey @Sednaoui, did you find a solution? I have the same issue for bundle calls.

**Comment by boidushya** [comment]
@Sednaoui For an advanced use case, have you tried using [EthereumProvider](https://docs.walletconnect.com/2.0/web/providers/ethereum) instead?
This lets you use RPC methods/request directly like so
```js
const result = await provider.request({ method: 'eth_requestAccounts' })
```
and supports passing `params` as well

If this doesn't suit your requirements, We'd suggest reaching out to [wagmi](https://github.com/wagmi-dev/wagmi) for this feature request

**Reply by Sednaoui** [reply]
Thank you @boidushya for the clarification. I understand now I can't use the provider in web3modal as it relies on wagmi connectors. 

I am trying to get this working web3modal in particular since the dapps who are trying to send custom requests already uses web3modal. I am going to ask wagmi in that case.
